### ISSUE TITLE
Authentication fails with custom kubectl plugin

### ISSUE THREAD

**Issue by Razz4780** [metalbear-co/mirrord#2023]
### Bug Description

`kube-rs` client fails to run custom authentication plugin. The plugin reads `KUBERNETES_EXEC_INFO` env var, which is a JSON-serialized object with two fields: `kind` and `apiVersion`. The plugin fails, because `kube-rs` client always sets `kind` to `null` when executing the command.

### Steps to Reproduce

Probably difficult, as we don't have the plugin.

### Relevant Logs

```shell
(async-tasks) ❯ mirrord exec --config-file .mirrord/config.json "celery -A services.feedback.main worker --loglevel=INFO"                                                                                  async-tasks -> main ? $
When targeting multi-pod deployments, mirrord impersonates the first pod in the deployment.
Support for multi-pod impersonation requires the mirrord operator, which is part of mirrord for Teams.
To try it out, join the waitlist with `mirrord waitlist <email address>`, or at this link: https://metalbear.co/#waitlist-form
  x mirrord exec
    ✓ Running on latest (3.72.1)!
    x preparing to launch process
      ✓ layer extracted
      x unable to check if operator exists, probably due to RBAC                                                                                                                                                                   Error:   × Failed to create Kubernetes API. KubeError(
  │     Auth(
  │         AuthExecRun {
  │             cmd: "KUBERNETES_EXEC_INFO=\"{\\\"kind\\\":null,\\\"apiVersion\\\":\\\"client.authentication.k8s.io/v1\\\",\\\"spec\\\":{\\\"interactive\\\":true}}\" \"poyo\" \"k8s\" \"auth\"",
  │             status: ExitStatus(
  │                 unix_wait_status(
  │                     256,
  │                 ),
  │             ),
  │             out: Output {
  │                 status: ExitStatus(
  │                     unix_wait_status(
  │                         256,
  │                     ),
  │                 ),
  │                 stdout: "",
  │                 stderr: "1:00PM ERR Error generating credentials error=\"error loading exec credential from environment: decode: Object 'Kind' is missing in '{\\\"kind\\\":null,\\\"apiVersion\\\":\\
  │ \"client.authentication.k8s.io/v1\\\",\\\"spec\\\":{\\\"interactive\\\":true}}'\"\n",
  │             },
  │         },
  │     ),
  │ )
  ├─▶ mirrord-layer: Kube failed with error `auth error: auth exec command 'KUBERNETES_EXEC_INFO="{\"kind\":null,\"apiVersion\":\"client.authentication.k8s.io/v1\",\"spec\":{\"interactive\":true}}" "poyo" "k8s" "auth"' failed
  │   with status exit status: 1: Output { status: ExitStatus(unix_wait_status(256)), stdout: "", stderr: "1:00PM ERR Error generating credentials error=\"error loading exec credential from environment: decode: Object 'Kind'
  │   is missing in '{\\\"kind\\\":null,\\\"apiVersion\\\":\\\"client.authentication.k8s.io/v1\\\",\\\"spec\\\":{\\\"interactive\\\":true}}'\"\n" }`!
  ├─▶ auth error: auth exec command 'KUBERNETES_EXEC_INFO="{\"kind\":null,\"apiVersion\":\"client.authentication.k8s.io/v1\",\"spec\":{\"interactive\":true}}" "poyo" "k8s" "auth"' failed with status exit status: 1: Output
  │   { status: ExitStatus(unix_wait_status(256)), stdout: "", stderr: "1:00PM ERR Error generating credentials error=\"error loading exec credential from environment: decode: Object 'Kind' is missing in '{\\\"kind\\\":null,\
  │   \\"apiVersion\\\":\\\"client.authentication.k8s.io/v1\\\",\\\"spec\\\":{\\\"interactive\\\":true}}'\"\n" }
  ╰─▶ auth exec command 'KUBERNETES_EXEC_INFO="{\"kind\":null,\"apiVersion\":\"client.authentication.k8s.io/v1\",\"spec\":{\"interactive\":true}}" "poyo" "k8s" "auth"' failed with status exit status: 1: Output { status:
      ExitStatus(unix_wait_status(256)), stdout: "", stderr: "1:00PM ERR Error generating credentials error=\"error loading exec credential from environment: decode: Object 'Kind' is missing in '{\\\"kind\\\":null,\\
      \"apiVersion\\\":\\\"client.authentication.k8s.io/v1\\\",\\\"spec\\\":{\\\"interactive\\\":true}}'\"\n" }
  help:
            Please check that Kubernetes is configured correctly.
            Test your connection with `kubectl get pods`.


        - If you're still stuck:

        >> Please open a new bug report at https://github.com/metalbear-co/mirrord/issues/new/choose

        >> Or join our Discord https://discord.gg/metalbear and request help in #mirrord-help

        >> Or email us at hi@metalbear.co
```

**Comment by aviramha** [comment]
Did we create an upstream issue?

**Comment by Razz4780** [comment]
> Did we create an upstream issue?

Creating one now

**Comment by Razz4780** [comment]
Resolved with #2027

**Comment by josh-bell-colab** [comment]
Can confirm fixed!

Thank you 🥳
